### PR TITLE
Changes to support e2e (BDD) testing using httptest server for projects routes and other changes

### PR DESCRIPTION
**Pull request by rqc**
**This pull request contains considerable changes**

Major  changes
============
- Changed standard port for running shipyard from 8888 (just in docker-compose.yml) to 8082 as per issue with AWS
- Re-integrated client libraries done by team and added a couple more to support new e2e workflows.
- Refactored Shipyard API initialization flow to split it into Setup() and Run() which allows us to test an API object better.
- Implemented e2e tests for Projects and nested ProjectImages routes with httptest.NewServer() by creating a real instance of the Manager and Api objects (real database conn) and initializing GorillaMux and Negroni middleware.
- New requirement for e2e tests: Need to specify SHIPYARD_RETHINKDB_URI env var when running Rethink on a uri:host different than rethinkdb:28015 (this is the default in the linked container).
- Removed the Godeps/_workspace in favor of the /vendor best practice for Golang. If you are using Go1.5 you need to export GO15VENDOREXPERIMENT=1 in your shell to make sure that it works consistently.
- Godeps is no longer needed in the Makefile recipes, so they were removed. Optional, in dev environment, you might want to use "godeps restore" with the GO15VENDOREXPERIMENT flag set to install the correct versions in your gopath.
- Removed references to proxies in the Dockerfile.build in favor of AWS deployments.

Minor changes
============
- Formatting changes (automatic by applying go fmt in the project)
- Added convenience recipes to Makefile, just to speed up e2e testing and dev in dev env.

### CODE REVIEW

File: client/projects.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	}
+	return images, resp.StatusCode, nil
+}
+
+func GetProjectImage(authHeader, url, projectId string, imageId string) (model.Image, int, error) {
+	var image model.Image
+	resp, err := sendRequest(authHeader, "GET", fmt.Sprintf("%s/api/projects/%s/images/%s", url, projectId, imageId), "")
+	if err != nil {
+		return image, resp.StatusCode, err
+	} else {
```
Comment (JonathanRosado) [inline client/projects.go]: No need for an else block

File: client/images.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
+package ilm_client
```
Comment (JonathanRosado) [inline client/images.go]: Everything seems OK

**Comment by gmatoshp** [comment]
Looks good to me too

File: client/projects.go (hunk 1 above)
Comment (rqc) [inline client/projects.go]: thanks, that's a good catch. And after removing those, it actually uncovered a bug in the `client` and in the test suite. Thanks!

**Comment by rqc** [comment]
@JonathanRosado @gmatoshp thanks! I will merge now.

**Pull request merged**